### PR TITLE
feat: Add endpoint for erasing projects permanently (DEV-3680)

### PR DESCRIPTION
**Pull request by seakayone**
- **add delete endpooint**
- **add config to disable erasing projects**

### CODE REVIEW

File: src/test/scala/swiss/dasch/api/ProjectsEndpointSpec.scala [hunk 1]
```diff
@@ -279,10 +282,32 @@ object ProjectsEndpointSpec extends ZIOSpecDefault {
...
+          val req = Request
+            .delete(URL(Root / "projects" / s"${shortcode.value}" / "erase"))
+            .addHeader("Authorization", "Bearer fakeToken")
+          for {
+            res              <- executeRequest(req)
+            folderWasRemoved <- StorageService.getProjectFolder(shortcode).flatMap(Files.exists(_)).negate
+          } yield assertTrue(res.status == Status.Ok, folderWasRemoved)
+        }
+    },
+  )
```
Comment (BalduinLandolt) [inline src/test/scala/swiss/dasch/api/ProjectsEndpointSpec.scala:304]: nitpick: I suppose you could also add a test to check that it doesn't work when allowEarseProjects is false... but maybe not needed for such basic logic

**Review by BalduinLandolt — APPROVED** [review]
nice, that was quick!

File: src/test/scala/swiss/dasch/api/ProjectsEndpointSpec.scala (hunk 1 above)
Comment (seakayone) [reply src/test/scala/swiss/dasch/api/ProjectsEndpointSpec.scala:304]: Unfortunately I do not have an idea how to write a simple spec which would only exchange the `Features` layer without introducing lots of duplication in the test code.

TLDR; Can do … though I am not sure if that test would be worth it.

File: src/main/scala/swiss/dasch/api/ProjectsEndpoints.scala [hunk 2]
```diff
@@ -166,6 +166,16 @@ final case class ProjectsEndpoints(base: BaseEndpoints) {
     .tag(projects)
     .description("Authorization: read:project:1234 scope required.")
 
+  val deleteProjectsErase = base.secureEndpoint.delete
```
Comment (mpro7) [inline src/main/scala/swiss/dasch/api/ProjectsEndpoints.scala:169]: instead of `deleteProjectsErase` I would use `deleteProjectsPermamently`. I think we didn't agree on naming convention here yet, however my proposal sounds a bit more natural. Applying this suggestion would trigger name change for related methods.

File: src/main/scala/swiss/dasch/api/ProjectsEndpointsHandler.scala [hunk 3]
```diff
@@ -73,6 +76,24 @@ final case class ProjectsEndpointsHandler(
...
+          projectService.findProject(shortcode).some.mapError(projectNotFoundOrServerError(_, shortcode)) *> {
+            if (features.allowEraseProject) {
+              projectService
+                .deleteProject(shortcode)
+                .mapBoth(
+                  InternalServerError(_),
+                  _ => ProjectResponse.from(shortcode),
+                )
+            } else {
+              ZIO.fail(Unauthorized("Project erasure is not enabled."))
```
Comment (mpro7) [inline src/main/scala/swiss/dasch/api/ProjectsEndpointsHandler.scala:92]: Question: According to the error message, should this be `Unauthorized` error type or rather something different? If you decide to apply above method name change, consider also the adjustment of the message to something like `Permanent deletion of projects is not enabled.` This should be rather plural, because we allow deletion of all projects or not at all.

**Review by mpro7 — APPROVED** [review]
Looks good, please check the suggestions below:

File: src/main/scala/swiss/dasch/api/ProjectsEndpoints.scala (hunk 2 above)
Comment (seakayone) [reply src/main/scala/swiss/dasch/api/ProjectsEndpoints.scala:169]: I am following the `methodPath` naming convention as all the other `val`s. Are you suggesting to change the path as well?

File: src/main/scala/swiss/dasch/api/ProjectsEndpointsHandler.scala (hunk 3 above)
Comment (seakayone) [reply src/main/scala/swiss/dasch/api/ProjectsEndpointsHandler.scala:92]: Maybe `Forbidden` is more suitable; I am happy for any suggestion of a better http status code?

Comment (seakayone) [reply src/main/scala/swiss/dasch/api/ProjectsEndpointsHandler.scala]: How about: https://github.com/dasch-swiss/dsp-ingest/pull/218/commits/0bc2dcea7a7f78b3920c589b270dddb4600a2df4 ?

File: src/main/scala/swiss/dasch/api/ProjectsEndpoints.scala (hunk 2 above)
Comment (siers) [reply src/main/scala/swiss/dasch/api/ProjectsEndpoints.scala:169]: A data recovery service blog post does indeed suggest erase is meant to signify permanent deletion, there were others. I suppose then it is good as it is.

https://www.stellarinfo.com/blog/what-is-the-difference-between-file-erase-and-file-delete/

**Review by siers — APPROVED** [review]

File: src/main/scala/swiss/dasch/api/ProjectsEndpoints.scala (hunk 2 above)
Comment (mpro7) [reply src/main/scala/swiss/dasch/api/ProjectsEndpoints.scala:169]: Thanks for the link. Whatever will remain here, it needs to be explicitly passed to the other team members to make the clear distinction between deactivation and permanent deletion of project.
